src/h2olog/misc/gen_raw_tracer.py

```diff
@@ -733,15 +730,21 @@ def generate_cplusplus(context, output_file):
 
 """ % (build_typedef_for_cplusplus(), build_bpf_header_generator(), event_id_t_decl, event_t_decl, usdts_def, handle_event_func, bpf))
 
+def usage():
+    print("usage: %s output_file d_files..." % sys.argv[0])
+    sys.exit(1)
+
 
 def main():
-  try:
-    (_, h2o_dir, output_file) = sys.argv
-  except:
-    print("usage: %s h2o_dir output_file" % sys.argv[0])
-    sys.exit(1)
+  output_file = sys.argv[1]
```

Comment:
Python raises `IndexError` exception if `len(sys.argv) < 2`. I suppose you should check `sys.argv` length  before accessing list or wrap `try-catch` block

```python
if len(sys.argv) < 3:
    usage()

output_file, *d_files = sys.argv[1:]
```

Reply:
Oops. Thanks for your suggestion. Applied in 7f3668e4d3a599809172c94d7255de79bed219f6